collector/manager/metrics_collector_manager.go

```diff
@@ -151,20 +151,33 @@ func (mcm *MetricsCollectorManager) declareMetricDefinitions(metricDetails []col
 
 	metricDefs := make([]hmetrics.MetricDefinition, len(endpoint.Metrics))
 
-	// For each metric in the endpoint, create a metric def for it.
-	// Notice metric tags override endpoint tags which override global tags.
-	// Do NOT allow pods to use agent environment variables since agent env vars may contain
-	// sensitive data (such as passwords). Only the global agent config can define tags
-	// with env var tokens.
-	globalTags := mcm.Config.Collector.Tags.ExpandTokens(true, additionalEnv)
-	endpointTags := endpoint.Tags.ExpandTokens(false, additionalEnv)
-
-	// we need these to expand tokens in the IDs
-	mappingFunc := expand.MappingFunc(false, additionalEnv)
-	mappingFuncWithEnv := expand.MappingFunc(true, additionalEnv)
-
 	for i, metric := range endpoint.Metrics {
 
+		// Define additional envvars with pod specific data for use in replacing ${env} tokens in tags.
+		env := map[string]string{
+			"METRIC:name":     	metric.Name,
+			"METRIC:id":	   	metric.ID,
+			"METRIC:units":	    	metric.Units,
+			"METRIC:description":	metric.Description,
+			"METRIC:type":		metric.Type,
```

Comment:
The "description" and "type" here is probably not what you think it is.  These are the explicit description and type values set by the user in the endpoint config YAML (it is a way for the user to define them for Jolokia endpoints and a way for the user to provide his own description to override what was obtained from the endpoint - aka from prometheus). But for prometheus, these are going to be blank most of the time - the actual description and type that prometheus gives us are in the MetricDetails (see below - lines 192 and 195). For Jolokia, this might also be the case, I'm going to look to see if JMX has the ability to extract details like this just like  prometheus.